docs/source/guides/profiling.rst

```diff
@@ -4,12 +4,63 @@
 Profiling
 ================================================================================
 
-Profiling evaluates a data asset to create candidate expectations that describe the data. Profiling helps build
```

Comment:
1. Brought back the screenshots of the profiling results rendered into HTML to make sure that users see the value up front.
2. Moved the text in a way that the initial explanation of profiling is likely to be familiar to users. The explanation of how profiling is related to expectations is moved into "How Are Expectations And Profiling Related?" section
3. Updated the description of running profiling in CLI to reflect latest changes.